### PR TITLE
add bash completion for install+remove+update commands (bnc#928650)

### PR DESCRIPTION
**Pull request by bmwiedemann**
https://bugzilla.opensuse.org/show_bug.cgi?id=928650

### CODE REVIEW

**Comment by bmwiedemann** [comment]
After a discussion, we concluded that it would be best, if libzypp or repo2solv could produce a per-repo cache file after a refresh. The cache file would contain such lines:
```
bash-4.2-75.3.1.x86_64
rpm-4.11.3-12.1.x86_64
```

**Comment by mlandres** [comment]
Can we include ```{patch,pattern,product}-info``` commands to provide a result similar to #50?

**Comment by bmwiedemann** [comment]
added those 3 ... they are somewhat slow though (and I dont want to spend time optimizing with caches if we are changing this part soon)

**Comment by mlandres** [comment]
libzypp-15.1.3 (available in obs://zypp:HEAD) supports writing solv.idx files to speed up completion.

File: src/bash-completion.sh [hunk 1]
```diff
@@ -23,8 +22,46 @@ _strip()
...
 
+_get_idx_files()
+{
+	local d=/var/cache/zypp/solv
+	local idx=$d/*/solv.idx
+	# this part covers the case
+	# where no idx files have been created by libzypp yet
+	# TODO: drop in 2019
+        set +o noglob
+	if echo $idx | grep -q -F '*' ; then
```
Comment (bmwiedemann) [inline src/bash-completion.sh]: This part seems to be a bit fragile as there might have been zypper ref calls before the first completion. Also only works for root user.

File: src/bash-completion.sh [hunk 2]
```diff
@@ -23,8 +22,46 @@ _strip()
...
 
+_get_idx_files()
+{
+	local d=/var/cache/zypp/solv
+	local idx=$d/*/solv.idx
+	# this part covers the case
+	# where no idx files have been created by libzypp yet
+	# TODO: drop in 2019
+	set +o noglob
+	if echo $idx | grep -q -F '*' ; then
```
Comment (bmwiedemann) [inline src/bash-completion.sh]: This part seems to be a bit fragile as there might have been zypper ref calls before the first completion. Also only works for root user.

File: src/bash-completion.sh (hunk 1 above)
Comment (mlandres) [inline src/bash-completion.sh]: You can't silently remove the solv files.
Existing solv files may belong to volatile media (CD/DVD/USB), so you'd enforce zypper asking to insert the media again. I can try to create missing idx files in zypp even if the solv file is up-to-date.

File: src/bash-completion.sh (hunk 2 above)
Comment (bmwiedemann) [inline src/bash-completion.sh]: With https://github.com/openSUSE/libzypp/pull/44 we could drop this ugly workaround.

**Comment by bmwiedemann** [comment]
Updated the code to not have any ugly workarounds anymore. IMHO it is ready to be merged.

**Pull request merged**